pkg/pillar/zedcloud/authen.go

```diff
@@ -0,0 +1,508 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Common code to communicate to zedcloud
+
+package zedcloud
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	zauth "github.com/lf-edge/eve/api/go/auth"
+	zcert "github.com/lf-edge/eve/api/go/certs"
+	etpm "github.com/lf-edge/eve/pkg/pillar/evetpm"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"math/big"
+	"mime"
+	"net/http"
+	"os"
+)
+
+var onBoardCert, deviceCert *tls.Certificate
+var serverLeafCert *x509.Certificate
+var deviceCertHash, onBoardCertHash, cloudCertHash []byte
+var onBoardCertBytes []byte
```

Comment:
In the next sprint we should put these in zedcloudContext and add a zedcloudContext.NewContext() function which can return an initialized struct.

Reply:
ok.